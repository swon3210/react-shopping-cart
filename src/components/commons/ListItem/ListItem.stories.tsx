import ListItem, { Props } from './ListItem';

export default {
  title: 'Components/Commons/ListItem',
  component: ListItem,
  argTypes: {},
};

const Template = (args: Props) => <ListItem {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  thumbnail:
    'https://images.unsplash.com/photo-1589365278144-c9e705f843ba?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=634&q=80',
  name: '상품 이름',
  price: '10,000',
  quantity: '3',
  size: 'SM',
};
