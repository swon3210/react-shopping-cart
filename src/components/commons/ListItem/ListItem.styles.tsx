import styled from 'styled-components';
import { Props } from './ListItem';
import { COLORS } from '../../../constants/colors';

type ItemProps = Pick<Props, 'size'>;

const THUMBNAIL_SIZE = {
  SM: {
    width: '120px',
    height: '120px',
  },
  MD: {
    width: '142px',
    height: '142px',
  },
};

const PRICE_SIZE = {
  SM: {
    marginBottom: '18px',
  },
  MD: {
    marginBottom: '25px',
  },
};

const PRICE_COLOR = {
  SM: {
    color: COLORS.BLACK_500,
  },
  MD: {
    color: COLORS.GRAY_700,
  },
};

export const ListItem = styled.li`
  display: inline-flex;
`;

export const Thumbnail = styled.img<{ size: ItemProps['size'] }>(({ size }) => ({
  objectFit: 'cover',
  objectPosition: 'center',
  marginRight: '18px',
  ...THUMBNAIL_SIZE[size],
}));

export const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Name = styled.span<{ size: ItemProps['size']; theme: Object }>(({ size, theme }) => ({
  fontSize: '20px',
  color: theme.TEXT_COLOR,
  letterSpacing: '0.5px',
  ...PRICE_SIZE[size],
}));

export const Price = styled.span<{ size: ItemProps['size']; theme: Object }>(({ size, theme }) => ({
  letterSpacing: '0.5px',
  ...PRICE_COLOR[size],
}));
