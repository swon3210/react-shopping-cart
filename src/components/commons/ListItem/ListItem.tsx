import * as Styled from './ListItem.styles';
import noImagePNG from '../../../assets/images/no-image.png';

export interface Props {
  size: 'SM' | 'MD';
  thumbnail?: string;
  name: string;
  price: string;
  quantity: string;
}

const ListItem = ({ size = 'SM', thumbnail = noImagePNG, name, price, quantity }: Props) => {
  return (
    <Styled.ListItem>
      <Styled.Thumbnail size={size} src={thumbnail} />
      <Styled.InfoWrapper>
        <Styled.Name size={size}>{name}</Styled.Name>
        <Styled.Price size={size}>
          {price}원 / 수량 : {quantity}개
        </Styled.Price>
      </Styled.InfoWrapper>
    </Styled.ListItem>
  );
};

export default ListItem;
