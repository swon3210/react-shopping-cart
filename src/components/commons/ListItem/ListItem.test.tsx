import ListItem from './ListItem';
import { render } from '@testing-library/react';

describe('ListItem Component', () => {
  it('ListItem Snapshot', () => {
    const listItemUtil = render(<ListItem name="이름" price="2000" quantity="2" size="MD" />);
    expect(listItemUtil.asFragment()).toMatchSnapshot();
  });
});
