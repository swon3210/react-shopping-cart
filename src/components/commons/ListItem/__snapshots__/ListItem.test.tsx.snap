// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component ListItem Snapshot 1`] = `
<DocumentFragment>
  <li
    class="sc-bdvvaa hGgTo"
  >
    <img
      class="sc-gsDJrp iBUonM"
      src="no-image.png"
    />
    <div
      class="sc-dkPtyc cJFzIl"
    >
      <span
        class="sc-hKwCoD dFBKqs"
      >
        이름
      </span>
      <span
        class="sc-eCImvq jrtxqM"
      >
        2000원 / 수량 : 2개
      </span>
    </div>
  </li>
</DocumentFragment>
`;
